lib/aspects/canary-aspect.ts

```diff
@@ -0,0 +1,41 @@
+import { IAspect, IConstruct, Stack, CfnResource, CfnOutput, Construct } from "@aws-cdk/core";
+import { BaseLoadBalancer } from "@aws-cdk/aws-elasticloadbalancingv2";
+import { Integration } from '@aws-cdk/aws-apigateway';
+import { Canary } from "../canaries/canary";
+import { CdkStage } from "../proposed_api/cdk-pipeline";
+
+export class CanaryAspect implements IAspect {
+  _canaryStack: Stack;
+
+  constructor(private readonly scope: Construct, private readonly stage: CdkStage) {}
+
+  public visit(node: IConstruct): void {
+    if (Stack.isStack(node)) {
+      const children = node.node.children;
+      children.forEach( ch => {
+        // we can add more types here, APIGateway etc.
+        if (ch instanceof BaseLoadBalancer) { 
+          new Canary(this.canaryStack, `Canary-${ch.node.uniqueId}`, {
+            url: ch.loadBalancerDnsName
+          });
+          (node as Stack).addDependency(this.canaryStack);
```

Comment:
Doesn't this lead to a circular dependency? `canaryStack` will already depend on `node` because `node` will export a Load Balancer DNS name.